common/src/main/java/io/netty/util/internal/RecyclableArrayList.java

```diff
@@ -119,7 +135,16 @@ public Object set(int index, Object element) {
         if (element == null) {
             throw new NullPointerException("element");
         }
-        return super.set(index, element);
+        Object old = super.set(index, element);
+        added = true;
+        return old;
+    }
+
+    /**
+     * Returns {@code true} if any elements where added or set. This will be reset once {@link #recycle()} was called.
```

Comment:
Maybe override `clear` and set `added = false`?

Would `modified` be a better name to use than `added`?